tusclient/fingerprint/fingerprint.py

```diff
@@ -22,9 +23,11 @@ def get_fingerprint(self, fs: IO):
         hasher = hashlib.md5()
         # we encode the content to avoid python 3 uncicode errors
         buf = self._encode_data(fs.read(self.BLOCK_SIZE))
-        while buf:
-            hasher.update(buf)
-            buf = fs.read(self.BLOCK_SIZE)
+        hasher.update(buf)
+        # add in the file size to minimize chances of collision
+        fs.seek(0, os.SEEK_END)
+        file_size = fs.tell()
+        hasher.update(str(file_size).encode())
         return 'md5:' + hasher.hexdigest()
```

Comment:
So what I mean is to do something like this instead:
```python
return f'{file_size}--md5:' + hasher.hexdigest()
```
So it won't be part of the checksum calculation, instead it is part of the final string

Reply:
We can do it that way, I salted it in (poorly, by forgetting to encode the integer)

Reply:
I can squash the history too so its not so circuitous if you want.

Reply:
I can squash right before merging via GitHub UI, however, I think now we need to fix the [broken test](https://travis-ci.org/github/tus/tus-py-client/jobs/689381101). I think we need to edit [this file](https://github.com/tus/tus-py-client/blob/master/tests/storage_file) to use the new fingerprint value.

Reply:
All set, that one also failed locally so we shouldn't see another CI fail.